libs/ledger-live-common/src/families/cosmos/react.ts

```diff
@@ -207,13 +194,11 @@ export function useLedgerFirstShuffledValidatorsCosmosFamily(
   currencyName: string,
   searchInput?: string
 ): CosmosValidatorItem[] {
-  let data;
   let ledgerValidatorAddress;
+  const data = getCurrentCosmosPreloadData();
   if (currencyName == "osmosis") {
```

Comment:
Same here. Maybe validators could be maintained in a Map<currency_id, validator_address>

Reply:
Fixed